packages/angular_devkit/build_angular/src/builders/browser-esbuild/sass-plugin.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import type { Plugin, PluginBuild } from 'esbuild';
+import type { LegacyResult } from 'sass';
+import { SassWorkerImplementation } from '../../sass/sass-service';
+
+export function createSassPlugin(options: { sourcemap: boolean; includePaths?: string[] }): Plugin {
+  return {
+    name: 'angular-sass',
+    setup(build: PluginBuild): void {
+      let sass: SassWorkerImplementation;
+
+      build.onStart(() => {
+        sass = new SassWorkerImplementation();
+      });
+
+      build.onEnd(() => {
+        sass?.close();
+      });
+
+      build.onLoad({ filter: /\.s[ac]ss$/ }, async (args) => {
```

Comment:
Can you maybe add a comment that the current implementation would not watch files when there are errors during the first run.  Possible this could be tacked through some custom importer.